easybuild/framework/easyblock.py

```diff
@@ -2378,6 +2378,14 @@ def fetch_step(self, skip_checksums=False):
                 raise EasyBuildError("EasyBuild-version %s is newer than the currently running one. Aborting!",
                                      easybuild_version)
 
+        source_deps = self.cfg['source_deps']
+        pre_fetch_env = None
+        # load modules for source dependencies (if any)
+        if source_deps:
+            pre_fetch_env = copy.deepcopy(os.environ)
+            source_deps_mod_names = [d['short_mod_name'] for d in source_deps]
+            self.modules_tool.load(source_deps_mod_names)
```

Comment:
@PetrKralCZ Before trying to load modules, we need to take into account that `self.modules_tool` may be a `NoModulesTool` instance (that will be the case when `eb --fetch` is used).

If so, then we need to re-initialize `self.modules_tool` to a proper `ModulesTool` instance (via the `modules_tool` function available in `tools/config.py`).

This will need changes in `options.py` too though, because there we're now overwriting `self.options.modules_tool` with `None`, we also need to keep track of the original value so we can then pass it to the `modules_tool` function (which will also need to be adjusted)...